### PR TITLE
Added create reactions group

### PR DESCRIPTION
**Pull request by maybee16**
https://trello.com/c/Jd8JE294/117-reactionservice

### CODE REVIEW

File: src/ReactionService.Business/Commands/Reaction/CreateReactionCommand.cs [hunk 1]
```diff
@@ -71,21 +71,21 @@ public class CreateReactionCommand : ICreateReactionCommand
...
 
     if (!isResizeSuccess)
     {
       response.Errors.Add("Resize operation have been failed");
 
-      if (Convert.FromBase64String(request.Content).Length / 1000 > 10)
+      if (Convert.FromBase64String(request.Image.Content).Length / 1000 > 10)
       {
-        (bool isCompressSuccess, imageContent, request.Extension) = await _imageCompressHelper.CompressAsync(request.Content, request.Extension, 10);
+        (bool isCompressSuccess, imageContent, request.Image.Extension) = await _imageCompressHelper.CompressAsync(request.Image.Content, request.Image.Extension, 10);
```
Comment (vorobevamv) [inline src/ReactionService.Business/Commands/Reaction/CreateReactionCommand.cs]: move 'await...' to next line

File: src/ReactionService.Business/Commands/Reaction/CreateReactionCommand.cs [hunk 2]
```diff
@@ -71,21 +71,21 @@ public class CreateReactionCommand : ICreateReactionCommand
...
       }
     }
 
     if ((imageContent is not null) && (Convert.FromBase64String(imageContent).Length / 1000 > 10))
     {
-      (bool isCompressSuccess, imageContent, request.Extension) = await _imageCompressHelper.CompressAsync(imageContent, request.Extension, 10);
+      (bool isCompressSuccess, imageContent, request.Image.Extension) = await _imageCompressHelper.CompressAsync(imageContent, request.Image.Extension, 10);
```
Comment (vorobevamv) [inline src/ReactionService.Business/Commands/Reaction/CreateReactionCommand.cs]: move 'await...' to next line

File: src/ReactionService.Business/Commands/Reaction/CreateReactionCommand.cs [hunk 3]
```diff
@@ -71,21 +71,21 @@ public class CreateReactionCommand : ICreateReactionCommand
 
     OperationResultResponse<Guid?> response = new();
 
-    (bool isResizeSuccess, string imageContent, request.Extension) = await _imageResizeHelper.ResizeAsync(request.Content, request.Extension, 24);
+    (bool isResizeSuccess, string imageContent, request.Image.Extension) = await _imageResizeHelper.ResizeAsync(request.Image.Content, request.Image.Extension, 24);
```
Comment (vorobevamv) [inline src/ReactionService.Business/Commands/Reaction/CreateReactionCommand.cs]: move 'await...' to next line

File: src/ReactionService.Business/Commands/ReactionsGroup/CreateReactionsGroupCommand.cs [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        validationResult.Errors.Select(x => x.ErrorMessage).ToList());
+    }
+
+    OperationResultResponse<Guid?> response = new();
+
+    List<DbReaction> dbReactions = new();
+
+    foreach (CreateReactionRequest reaction in request.Reactions)
+    {
+      (bool isResizeSuccess, string imageContent, reaction.Image.Extension) = await _imageResizeHelper.ResizeAsync(reaction.Image.Content, reaction.Image.Extension, 24);
```
Comment (TIove) [inline src/ReactionService.Business/Commands/ReactionsGroup/CreateReactionsGroupCommand.cs]: :84 :92 :98 lines are too long. Set enter after '=' for example.

**Review by TIove — APPROVED** [review]

File: src/ReactionService.Business/Commands/ReactionsGroup/CreateReactionsGroupCommand.cs (hunk 4 above)
Comment (maybee16) [reply src/ReactionService.Business/Commands/ReactionsGroup/CreateReactionsGroupCommand.cs]: Done

File: src/ReactionService.Business/Commands/Reaction/CreateReactionCommand.cs (hunk 1 above)
Comment (maybee16) [reply src/ReactionService.Business/Commands/Reaction/CreateReactionCommand.cs]: Done

File: src/ReactionService.Business/Commands/Reaction/CreateReactionCommand.cs (hunk 2 above)
Comment (maybee16) [reply src/ReactionService.Business/Commands/Reaction/CreateReactionCommand.cs]: Done

File: src/ReactionService.Business/Commands/Reaction/CreateReactionCommand.cs (hunk 3 above)
Comment (maybee16) [reply src/ReactionService.Business/Commands/Reaction/CreateReactionCommand.cs]: Done

**Review by vorobevamv — APPROVED** [review]

**Review by AlfaRedDog — APPROVED** [review]

File: src/ReactionService.Data/ReactionsGroupRepository.cs [hunk 5]
```diff
@@ -16,6 +19,49 @@ public class ReactionsGroupRepository : IReactionsGroupRepository
...
+  public Task<DbReactionsGroup> GetAsync(GetReactionsGroupFilter filter)
+  {
+    return filter is null
+      ? null
+      : CreateGetPredicates(filter).FirstOrDefaultAsync(x => x.Id == filter.ReactionsGroupId);
+  }
+
+  public Task<bool> DoesNameExist(string name)
+  {
+    return _provider.ReactionsGroups.AsNoTracking().AnyAsync(x => x.Name == name);
```
Comment (Vasis3038) [inline src/ReactionService.Data/ReactionsGroupRepository.cs]: Do we really need AsNoTracking() here?
Does Any() save anything in cache?

Comment (maybee16) [reply src/ReactionService.Data/ReactionsGroupRepository.cs]: No, it's not needed here. Done

**Review by Vasis3038 — APPROVED** [review]

**Pull request merged**